packages/devtools_app/test/memory_screen_test.dart

```diff
@@ -159,4 +183,86 @@ void main() {
+  testWidgetsWithWindowSize('heap tree view', windowSize,
+      (WidgetTester tester) async {
+    await pumpMemoryScreen(tester);
+
+    expect(find.byKey(HeapTreeViewState.snapshotButtonKey), findsOneWidget);
+    expect(
+      find.byKey(HeapTreeViewState.allocationMonitorResetKey),
+      findsOneWidget,
+    );
+    expect(
+      find.byKey(HeapTreeViewState.allocationMonitorResetKey),
+      findsOneWidget,
+    );
+
+    // Load canned data.
+    _setUpServiceManagerForMemory();
+
+    final data = controller.memoryTimeline.data;
+
+    // Total number of collected HeapSamples.
+    expect(data.length, 292);
+
+    // Number of VM GCs
+    final totalGCEvents = data.where((element) => element.isGC);
+    expect(totalGCEvents.length, 70);
+
+    // User initiated GCs
+    final totalUserGCEvents =
+        data.where((element) => element.memoryEventInfo.isEventGC);
+    expect(totalUserGCEvents.length, 2);
+
+    // User initiated Snapshots
+    final totalSnapshotEvents =
+        data.where((element) => element.memoryEventInfo.isEventSnapshot);
+    expect(totalSnapshotEvents.length, 2);
+
+    // Number of auto-Snapshots
+    final totalSnapshotAutoEvents =
+        data.where((element) => element.memoryEventInfo.isEventSnapshotAuto);
+    expect(totalSnapshotAutoEvents.length, 3);
+
+    // Total Allocation Monitor events (many are empty).
+    final totalAllocationMonitorEvents = data.where(
+        (element) => element.memoryEventInfo.isEventAllocationAccumulator);
+    expect(totalAllocationMonitorEvents.length, 285);
+
+    // Number of user initiated allocation monitors
+    final startMonitors = totalAllocationMonitorEvents.where(
+        (element) => element.memoryEventInfo.allocationAccumulator.isStart);
+    expect(startMonitors.length, 3);
+
+    // Number of accumulator resets
+    final resetMonitors = totalAllocationMonitorEvents.where(
+        (element) => element.memoryEventInfo.allocationAccumulator.isReset);
+    expect(resetMonitors.length, 2);
+
+    final interval1Min = MemoryController.displayIntervalToIntervalDurationInMs(
+        MemoryController.displayOneMinute);
+    expect(interval1Min, 60000);
+    final interval5Min = MemoryController.displayIntervalToIntervalDurationInMs(
+        MemoryController.displayFiveMinutes);
+    expect(interval5Min, 300000);
+
+    // TODO(terry): Check intervals and autosnapshot does it snapshot same points?
+    // TODO(terry): Simulate sample run of liveData filling up?
+
+    // Take a snapshot
+    await tester.tap(find.byKey(HeapTreeViewState.snapshotButtonKey));
+    await tester.pump();
+
+    final snapshotButton = tester
+        .widget<OutlineButton>(find.byKey(HeapTreeViewState.snapshotButtonKey));
+
+    expect(snapshotButton.enabled, isFalse);
+    await tester.pumpAndSettle(const Duration(seconds: 3));
+
+    expect(
+      controller.selectedSnapshotTimestamp.millisecondsSinceEpoch,
+      lessThan(DateTime.now().millisecondsSinceEpoch),
+    );
+  });
 }
```

Comment:
a golden image test of this would be easy to add and would help check more of the state.

Reply:
I'll add a golden image test post this PR that's a good idea.